straxen/plugins/event_processing.py

```diff
@@ -733,58 +733,175 @@ def cs2_to_e(self, x):
-                 ('shadow_distance', np.float32, 'distance to the s2 peak with max shadow [cm]')]
+        dtype = []
+        for sa, r in zip(['', 'alt_'], ['1st', '2nd']):
+            for s in ['s1', 's2']:
+                for p, si in zip(['s1', 's2', 's2_re'], ['s1', 's2', 'reordered s2']):
+                    dtype.append((('main ' + s + ' shadow from ' + si + ' casting ' + r + ' largest shadow [PE/ns]', s + '_' + sa + 'shadow_' + p), np.float32))
+                    dtype.append((('previous ' + si + ' area casting ' + r + ' largest shadow on ' + s + ' [PE]', s + '_' + sa + 'pre_area_' + p), np.float32))
+                    dtype.append((('time difference from ' + s + ' to the previous ' + si + ' casting ' + r + ' largest shadow [ns]', s + '_' + sa + 'shadow_dt_' + p), np.int64))
+            for p, si in zip(['s1', 's2', 's2_re'], ['s1', 's2', 'reordered s2']):
+                dtype.append((('event shadow from ' + si + ' casting ' + r + ' largest shadow [PE/ns]', sa + 'shadow_' + p), np.float32))
+                dtype.append((('previous ' + si + ' area casting ' + r + ' largest shadow  on the event [PE]', sa + 'pre_area_' + p), np.float32))
+                dtype.append((('time difference from the event to the previous ' + si + ' casting ' + r + ' largest shadow [ns]', sa + 'shadow_dt_' + p), np.int64))
+            for x in ['x', 'y']:
+                dtype.append(((x + ' of previous s2 peak casting ' + r + ' largest shadow [cm]', sa + 'pre_' + x + '_s2'), np.float32))
+                dtype.append(((x + ' of previous reordered s2 peak casting ' + r + ' largest shadow [cm]', sa + 'pre_' + x + '_s2_re'), np.float32))
+            dtype.append((('distance to the s2 peak with ' + r + ' largest shadow [cm]', sa + 'shadow_distance'), np.float32))
+            dtype.append((('distance to the reordered s2 peak with ' + r + ' largest shadow [cm]', sa + 'shadow_distance_re'), np.float32))
+            dtype.append((('previous ' + r + ' largest s2 shadow with position correlation PDF [PE/ns]', sa + 'shadow_s2_corr'), np.float32))
+            dtype.append((('previous ' + r + ' largest s2 shadow position correlation PDF', sa + 'shadow_s2_prob'), np.float32))
+        dtype.append((('index of the peak defining the event shadow', 'shadow_index'), np.int32))
+        for s in ['s1', 's2']:
+            for p, si in zip(['s1', 's2', 's2_re'], ['s1', 's2', 'reordered s2']):
+                dtype.append((('Nearest ' + si + ' delta t to ' + s, s + '_near_dt_' + p), np.int64))
+                dtype.append((('Sum of ' + si + ' shadow in time window casting on ' + s, s + '_shadow_sum_' + p), np.float32))
+        for p, si in zip(['s1', 's2', 's2_re'], ['s1', 's2', 'reordered s2']):
+            dtype.append((('Nearest ' + si + ' delta t to event', 'near_dt_' + p), np.int64))
+            dtype.append((('Sum of ' + si + ' shadow in time window casting on event', 'shadow_sum_' + p), np.float32))
```

Comment:
sorry, but to me it's quite confusing what is the difference between s1/s2, main_s1/s2, previous_s1/s2, reordered_s2, etc. Can you maybe make a plot which illustrates each variables such that people can easily judge?

Reply:
for example, I cannot judge what "s1_alt_pre_area_s1" means by just looking at the corresponding variable...